pyanaconda/core/configuration/storage_constraints.py

```diff
@@ -0,0 +1,159 @@
+    MD = DEVICE_TYPE_MD
+    PARTITION = DEVICE_TYPE_PARTITION
+    BTRFS = DEVICE_TYPE_BTRFS
+    DISK = DEVICE_TYPE_DISK
+    LVM_THINP = DEVICE_TYPE_LVM_THINP
+
+
+class StorageConstraints(Section):
+    """The Storage Constraints section."""
+
+    @property
+    def min_ram(self):
+        """Minimal size of the total memory."""
+        return self._get_option("min_ram", Size)
+
+    @property
+    def luks2_min_ram(self):
+        """Minimal size of the available memory for LUKS2."""
+        return self._get_option("luks2_min_ram", Size)
+
+    @property
+    def swap_is_recommended(self):
+        """Should we recommend to specify a swap partition?"""
+        return self._get_option("swap_is_recommended", bool)
+
+    @property
+    def min_partition_sizes(self):
+        """Recommended minimal sizes of partitions.
+
+        :return: a dictionary of partition sizes
+        """
+        return self._get_option("min_partition_sizes", self._convert_partition_sizes)
+
+    @property
+    def req_partition_sizes(self):
+        """Required minimal sizes of partitions.
+
+        :return: a dictionary of partition sizes
+        """
+        return self._get_option("req_partition_sizes", self._convert_partition_sizes)
+
+    def _convert_partition_sizes(self, value):
+        """Convert the given value into a dictionary of partition sizes."""
+        lines = value.split("\n")
+        sizes = {}
+
+        for line in lines:
+            if line:
+                mount_point, size = line.split(maxsplit=1)
+                sizes[mount_point] = Size(size)
+
+        return sizes
+
+    @property
+    def root_device_types(self):
+        """Allowed device types of the / partition if any.
+
+        Valid values:
+
+          0  LVM        Allow LVM.
+          1  MD         Allow RAID.
+          2  PARTITION  Allow standard partitions.
+          3  BTRFS      Allow Btrfs.
+          4  DISK       Allow disks.
+          5  LVM_THINP  Allow LVM Thin Provisioning.
+
+        :return: a set of device types
+        """
+        return self._get_option("root_device_types", self._convert_device_types)
+
+    def _convert_device_types(self, value):
+        """Convert the given value into a set of device types."""
+        return set(map(self._convert_device_type, value.split()))
+
+    def _convert_device_type(self, value):
+        """Convert the given value into a device type."""
+        try:
+            member = DeviceType.__members__[value]
+            return member.value
+        except KeyError:
+            pass
+
+        raise ValueError("'{}' is not a valid device typ".format(value))
+
+    @property
+    def must_be_on_linuxfs(self):
+        """Mount points that must be on a linux file system.
+
+        :return: a set of mount points
+        """
+        return set(self._get_option("must_be_on_linuxfs").split())
+
+    @property
+    def must_be_on_root(self):
+        """Mount points that must be on the / file system.
```

Comment:
Shouldn't it be:
``Mount points that must be on the / partition.``

IMHO FS is in most cases used for format used on the storage device but this is not what we are talking here.

Reply:
Fixed.

Reply:
I know I'm late to the party, but is device possibly a good word here?

Reply:
I have changed it to:
`Paths that must be directories on the / file system.`